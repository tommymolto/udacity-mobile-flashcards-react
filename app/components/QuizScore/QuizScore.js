import React from 'react';
import { Text, View } from 'react-native';
import { connect } from 'react-redux';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';
import SecondaryButton from '../Buttons/SecondaryButton';
import { resetQuiz } from '../../actions/quizActions';
import {
  clearLocalNotification,
  setLocalNotification,
} from '../../utils/helpers';
const StyledQuizScoreText = styled.Text`
  font-size: 20;
  padding-vertical: 15;
  padding-horizontal: 15;
  text-align: center;
`;

const QuizScore = ({
  totalQuizQuestions,
  userCorrectScore,
  userIncorrectScore,
  userResetQuiz, backToDeck,
  volta,
}) => {
  const dt = new Date();
  clearLocalNotification()
    .then(setLocalNotification(dt));
  return (
    <View>
      <StyledQuizScoreText>
        You got {userCorrectScore} right and {userIncorrectScore} wrong from a
        total of {totalQuizQuestions} questions.
      </StyledQuizScoreText>
      <SecondaryButton
        text="Reset Quiz?"
        onPress={() => {
          userResetQuiz();
        }}
      />
      <SecondaryButton
        text="Back to Deck"
        onPress={volta}
      />

    </View>
  );
};

QuizScore.propTypes = {
  totalQuizQuestions: PropTypes.number.isRequired,
  userCorrectScore: PropTypes.number.isRequired,
  userIncorrectScore: PropTypes.number.isRequired,
  userResetQuiz: PropTypes.func.isRequired,
  backToDeck: PropTypes.func.isRequired,
  volta: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  userQuizScore: state.quiz.quizNumbers.quizScore,
});

const mapDispatchToProps = dispatch => ({
  userResetQuiz: () => {
    dispatch(resetQuiz());
  },
  backToDeck: () => {
    this.volta();
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(QuizScore);
