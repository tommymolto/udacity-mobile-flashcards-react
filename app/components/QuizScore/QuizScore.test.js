/* global describe, it, expect */

import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import QuizScore from './QuizScore';
import storeConfig from '../../store/storeConfig';

describe('QuizScore', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Provider store={storeConfig}>
        <QuizScore
          totalQuizQuestions={2}
          userCorrectScore={1}
          userIncorrectScore={1}
          userResetQuiz={() => {}}
        />
              </Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
