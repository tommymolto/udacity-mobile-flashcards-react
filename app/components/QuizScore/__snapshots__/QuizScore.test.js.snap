// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizScore renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 20,
          "paddingHorizontal": 15,
          "paddingVertical": 15,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    You got 
    1
     right and 
    1
     wrong from a total of 
    2
     questions.
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "#FF2D40",
            "color": "#ffffff",
            "fontSize": 22,
            "marginLeft": 40,
            "marginRight": 40,
            "marginVertical": 10,
            "paddingBottom": 10,
            "paddingTop": 10,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Reset Quiz?
    </Text>
  </View>
</View>
`;
